### ISSUE TITLE
UI ignores "-bios" string stored in .ini files after running a game once during the same session when "skip_biosmenu" is turned on, reverts to default bios value in driver

### ISSUE THREAD

**Issue by ghost** [mamedev/mame#11313]
### MAME version

0.255

### System information

Windows 10 Home, 64-bit, version 22H2, USA, US English
Installed RAM 8.00 GB
AMD A6-5200 APU with Radeon(TM) HD Graphics 2.00 GHz

### INI configuration details

```INI
mame -cc

ui.ini only
skip_biosmenu             1
```

### Emulated system/software

UI only, neogeo/neogeo.cpp, igs/pgm.cpp, and sega/stv.cpp confirmed, possibly affects all systems with bios?

### Incorrect behaviour

Mame ignores bios settings in the .ini file during a UI session after first playing any game, then ending that emulation and starting a new emulation, but when the "Skip Bios selection menu" option is turned ON.  It can be the same game too.  Example:

1. I run the UI and select Street Fighter II to play.  After I'm done I hit escape and go back to the UI.  Now this game doesn't use a bios at all.
2. I select Knights of Valour, a pgm based game.  I have a kov.ini file with `bios v1` which equates `bios 1` as the default is v2 ("0").
3. I TAB to get to the BIOS selection menu in the UI while the game is up, and it lists "v2" and not what I have in the kov.ini file.

Another example:
1. I run the UI and select Knights of Valour (bios v1) first this time, check in the UI that it says "v1", then exit the game back to the UI.
2. I decide to play Knights of Valour again, checking the bios setting in the UI shows that it is now "v2".

### Expected behaviour

Turning on the Skip BIOS menu selection option should not make the UI ignore any bios settings in .ini files.

### Steps to reproduce

1. Double-click mame.exe.
2. Go to "General settings > Miscellaneous options > Skip BIOS selection menu" and turn that option ON.
3. Save the setting.
4. Run any game while still in the UI.
5. Exit that game.
6. Now run a game that has a specific non-default (not 0) bios setting in an .ini file.  Let's use kov with kov.ini.
7. Press TAB while the game is running and check the BIOS setting and you'll see it doesn't match the file.

### Additional details

I checked games as stand alone machines and not as a multi-slot setup, so I don't know how those are affected and that seems complicated to work out.  As noted above, I confirmed neogeo, pgm, and stv based games are affected, all arcade based.  I didn't check software lists, but maybe someone else can have fun with that.

I can't say for sure when this all started, but it has been going on for a while.  One day I'm looking at my nvram folder and saw some games had extra nvrams that I know should not, like Euro region for games I have set for only US or JP.

**Comment by Joejoe55** [comment]
So like the person above, I to have the same issue, went back and started testing from 255 down, and it works in 250 but not after, so started in 251 and up, also tested after 255 and has the issue, so last working is 250
sorry can't be of more help

**Comment by bojo5150** [comment]
I can also confirm this behavior in the newer releases

**Comment by EmuAl** [comment]
yes this seems to be a recent thing for me too

**Comment by Immersion95** [comment]
Same thing happening here and it's frustrating. Did you find any solution while waiting for an official fix ?

**Comment by Immersion95** [comment]
I've made a fix to this issue as it was really bothering me :p.

Can you try it and tell me if there is any regression ? It's on MAME 0.288.

https://github.com/Immersion95/mame/actions

**Comment by JimCarlTay** [comment]
Fixed in commit f6258eb, can now be closed as completed.

**Comment by Immersion95** [comment]
Yes it is, I think it's better to wait for the official release to close the 2 issues :).
